### DISCUSSION TITLE
Terminate TLS with two secrets for the same host

### DISCUSSION THREAD

**Discussion by vudex** [nginx/kubernetes-ingress#2019] [Q&A] [answered]
Hi. I'm trying to make my lab work, there are following prerequisites:

> Single host name test.example.com with two type of certs, RSA and EC based.
> With raw Nginx it is easily solvable with additional `ssl_certificate` and `ssl_certificate_key` directives, but I'm trying to make application work inside kubernetes with nginx ingress controller. 

The need of such separation is dictated by custom mobile devices, which dived in two groups - ones which work with rsa and ec algorithms and second that works only with EC algorithms.

First solution that comes in mind is to just terminate tls before kubernetes cluster with loadbalancer, but I'm curious is there any way to achieve this with ingress controller itself.

**Comment by pleshakov** [comment, marked as answer]
Hi @vudex The Ingress Controller doesn't support configuring two secrets directly. However, there is a workaround with snippets -- https://github.com/nginxinc/kubernetes-ingress/issues/1899#issuecomment-905952871

**Comment by shaun-nx** [comment]
Closing this as the original response was marked as the answer.
@vudex please feel free to re-open this discussion again if you feel it is necessary. Thanks!
